File: python/palindrome_spec.py
# Can you translate this driver code to unit tests?

import unittest
from palindrome import palindrome

class TestPalindrome(unittest.TestCase):

    def test_palindrome_words(self):
        self.assertTrue(palindrome('racecar'))
        self.assertTrue(palindrome('Noon'))
        self.assertTrue(palindrome('ciVic'))
        self.assertFalse(palindrome('nice'))
        self.assertTrue(palindrome(434))
        self.assertFalse(palindrome(123))
        self.assertFalse(palindrome('bomb'))

    def test_palindrome_sentences(self):
        self.assertTrue(palindrome('Sore was I ere I saw Eros.'))
        self.assertTrue(palindrome('A man, a plan, a canal -- Panama'))

if __name__ == '__main__':
    unittest.main()
#should any of these fail on purpose?

File: js/palindromeSpec.js
// Can you translate this driver code to unit tests?

const assert = require('assert');
const pal = require('./palindrome');

assert.strictEqual(pal.palindrome('racecar'), true);
assert.strictEqual(pal.palindrome('Noon'), true);
assert.strictEqual(pal.palindrome('ciVic'), true);
assert.strictEqual(pal.palindrome('nice'), false);
assert.strictEqual(pal.palindrome(434), true);
assert.strictEqual(pal.palindrome(123), false);

// console.log("The following should be true if you're trying to do the extra portion of this challenge");
assert.strictEqual(pal.palindrome("Sore was I ere I saw Eros."), true);
assert.strictEqual(pal.palindrome("A man, a plan, a canal -- Panama"), true);

console.log("All tests passed!");
